### ISSUE TITLE
Omit should throw error if not existing key try to remove

### ISSUE THREAD

**Issue by pakhuta** [microsoft/TypeScript#40262]
## Search Terms

 Omit inexistent key

## Suggestion
Let's consider next code where we omit inexistent key from interface `Todo` :
```ts
interface Todo {
    title: string;
    description: string;
    completed: boolean;
}

type TodoPreview = Omit<Todo, 'not_existed_key'>; // currently error is not thrown
```

## Use Cases
Let's imagine that previously interface `Todo` had property `not_existed_key` and we had variable such as
```ts
const todoItem: TodoPreview = {...}
```
but then `not_existed_key` was removed and developer forgot to remove `TodoPreview` because it became redundant.

## Examples

From my point of view will be better to have more strictly behavior of `Omit`.

## Checklist

My suggestion meets these guidelines:

* [ ] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by RyanCavanaugh** [comment]
See discussion at #30825

**Comment by pakhuta** [comment]
Thanks @RyanCavanaugh !
